jetty-core/jetty-http/src/test/java/org/eclipse/jetty/http/HttpCookieTest.java

```diff
@@ -94,34 +90,34 @@ public void testSetRFC2965Cookie() throws Exception
     {
         HttpCookie httpCookie;
 
-        httpCookie = new HttpCookie("null", null, null, null, -1, false, false, null, -1);
-        assertEquals("null=", httpCookie.getRFC2965SetCookie());
+        httpCookie = HttpCookie.from("null", null, -1, Map.of(HttpCookie.MAX_AGE_ATTRIBUTE, Long.toString(-1), HttpCookie.HTTP_ONLY_ATTRIBUTE, Boolean.toString(false), HttpCookie.SECURE_ATTRIBUTE, Boolean.toString(false)));
```

Comment:
Feels like this should have been a builder pattern.

``` java
httpCookie = HttpCookie.from("null", null)
   .maxAge(-1)
   .httpOnly(false)
   .secure(false)
   .build()
assertEquals("null=", HttpCookie.getRFC2965SetCookie(httpCookie));
```

or at least, this specific example should be using all defaults. (i would expect maxAge, httpOnly, and secure, to be defaulted where they wouldn't be set unless specified)

``` java
httpCookie = HttpCookie.from("null", null)
assertEquals("null=", HttpCookie.getRFC2965SetCookie(httpCookie));
```

Reply:
I was originally planning on having a Builder, but then worked out that the only places I would use it would be in tests.   
The Map.of is not pretty, but it is only test code.   So it seamed pointless to add a mechanism only for testing.